### PR TITLE
Add support for horizontal text scaling.

### PR DESCRIPTION
**Pull request by msbit**
Add methods and attrs for storing and inspecting horizontal text scaling references.

### CODE REVIEW

File: lib/pdf/inspector/text.rb [hunk 1]
```diff
@@ -62,6 +64,11 @@ def set_word_spacing(*params)
         @word_spacing << params[0]
       end
       
+      def set_horizontal_text_scaling(*params)
```
Comment (pointlessone) [inline lib/pdf/inspector/text.rb]: Why do you use catch all here instead of single parameter?

**Comment by pointlessone** [comment]
@msbit Thank you for all the contributions you've made recently.

Specs? ;)

File: lib/pdf/inspector/text.rb (hunk 1 above)
Comment (msbit) [inline lib/pdf/inspector/text.rb]: To be honest, for this, given that the PDF operators Tc (character spacing) and Tz (horizontal text scaling) seem to work very similarly, I've copied the method from set_character_spacing.

**Comment by msbit** [comment]
Can write some if you want, was taking my cue from the fact that there were none there already :smiley:

File: lib/pdf/inspector/text.rb (hunk 1 above)
Comment (pointlessone) [inline lib/pdf/inspector/text.rb]: I see. OK then. We'll refactor it later probably.

**Comment by pointlessone** [comment]
I know the coverage is not very good, to say the least. We should strive for better. ;)

**Comment by msbit** [comment]
So, as you may have guessed from my other PR on pdf-core, my motivation is to get horizontal text scaling going in prawn itself.

From my testing it's working, and in order to get the specs running for that change, pdf-inspector needed to be able to keep track of horizontal scaling changes coming in from pdf-reader.

I'm not, however, especially au fait with how pdf-inspector and pdf-reader work together, so I'm probably not the right person to _start_ getting the testing in place for this gem. I can definitely add to the specs once something is in place.

File: lib/pdf/inspector/text.rb (hunk 1 above)
Comment (msbit) [inline lib/pdf/inspector/text.rb]: I've been looking into writing specs for this, and I now understand that the reason why this method, and all other similar methods take the catch all is because it can be a variable argument list, as different PDF operators take different numbers of params. E.g., the `set_text_font_and_size` method (`Tm` PDF operator) takes two arguments.

**Comment by msbit** [comment]
Bit the bullet and wrote some specs for easily settable text attributes (character spacing and horizontal text scaling).

File: lib/pdf/inspector/text.rb (hunk 1 above)
Comment (pointlessone) [inline lib/pdf/inspector/text.rb]: That crossed my mind, actually. It may be the case for other operators. But as far as I can tell from PDF 1.7 spec `Tm` takes exactly 6 arguments. As well as other text-related operators take fixed number of arguments.

In this case we only use the first argument anyway (which, I suppose is an array of number). So why do we use more generic code and fiddle with argument extraction is we can have just one argument (or six)?

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
```
Comment (pointlessone) [inline Gemfile]: I understand the motivation but this seem to be unrelated to this PR.

Comment (msbit) [inline Gemfile]: I've removed that particular line as part of a follow up commit.
I was thinking though, did you mean just that line or adding Gemfile + bundler in general?

Comment (pointlessone) [inline Gemfile]: Gemfile and bundler.

Comment (msbit) [inline Gemfile]: :+1:

File: pdf-inspector.gemspec [hunk 3]
```diff
@@ -16,6 +16,8 @@ Gem::Specification.new do |spec|
                 "jimmy@deefa.com"]
   spec.licenses = %w(PRAWN GPL-2.0 GPL-3.0)
   spec.add_dependency('pdf-reader', '~>1.0')
+  spec.add_development_dependency('rspec')
+  spec.add_development_dependency('simplecov')
```
Comment (pointlessone) [inline pdf-inspector.gemspec]: Now, if we only could spare new deps… At least for now. ;)

Comment (msbit) [inline pdf-inspector.gemspec]: I suppose if I'm not building up a Gemfile nor a gem itself, there's no reason to have them there.

Comment (pointlessone) [inline pdf-inspector.gemspec]: I think, we can keep RSpec since we're using it anyway. But SimpleCov is a new dep that I would skip. At least in the context of this PR.

**Comment by pointlessone** [comment]
@msbit This gem is mostly used in specs elsewhere. Could you please reference relevant PRs on other gems if there are any? Just so that we knew we need to merge this one before the PRs that depend on it.

**Comment by msbit** [comment]
No outstanding PRs in other projects rely on this.

Once this and my two other PRs for pdf-core (https://github.com/prawnpdf/pdf-core/pull/26 and https://github.com/prawnpdf/pdf-core/pull/27) are tagged and versioned, I've got a branch on my fork of prawn that I'll put forward as a PR.

**Comment by msbit** [comment]
👍
